src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,51 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.health = 100;
+    this.name = name;
+
+    Animal.alive.push(this);
+  }
+
+  get isAlive() {
+    return this.health > 0;
+  }
+
+  applyDamage(healthPoints) {
+    this.health -= healthPoints;
+
+    if (this.health <= 0) {
+      Animal.cleanupDeadAnimals();
+    }
+  }
 }
 
+Animal.alive = [];
+Animal.cleanupDeadAnimals = () => {
+  Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
+};
```

Comment:
This cleanup method uses filter, which is good for performance as per the checklist. However, consider that this method will be called every time an animal's health reaches 0. Depending on the game logic, this could lead to performance issues if there are many animals and the method is called frequently. It might be more efficient to only clean up at certain intervals, or when the alive animals array reaches a certain size.